yt_dlp/extractor/ciscolive.py

```diff
@@ -2,144 +2,171 @@
+    IE_NAME = 'ciscolive:search'
+
+    _VALID_URL = r'https?://(?:www\.)?ciscolive\.com/on-demand/on-demand-library\.html\?[^#]+#/(?!session|video)[^/?#]*$'
     _TESTS = [{
-        'url': 'https://ciscolive.cisco.com/on-demand-library/?search.event=ciscoliveus2018&search.technicallevel=scpsSkillLevel_aintroductory&search.focus=scpsSessionFocus_designAndDeployment#/',
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?search.event=1737762187215001jsy4#/',
         'info_dict': {
-            'title': 'Search query',
+            'id': 'search',
         },
-        'playlist_count': 5,
-    }, {
-        'url': 'https://ciscolive.cisco.com/on-demand-library/?search.technology=scpsTechnology_applicationDevelopment&search.technology=scpsTechnology_ipv6&search.focus=scpsSessionFocus_troubleshootingTroubleshooting#/',
-        'only_matching': True,
+        'playlist_count': 500,
     }, {
-        'url': 'https://www.ciscolive.com/global/on-demand-library.html?search.technicallevel=scpsSkillLevel_aintroductory&search.event=ciscoliveemea2019&search.technology=scpsTechnology_dataCenter&search.focus=scpsSessionFocus_bestPractices#/',
-        'only_matching': True,
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?search.event=1707169032930001EEu2&search.technology=1538390420915002wPJx#/',
+        'info_dict': {
+            'id': 'search',
+        },
+        'playlist_count': 34,
     }]
 
-    @classmethod
-    def suitable(cls, url):
-        return False if CiscoLiveSessionIE.suitable(url) else super().suitable(url)
-
-    @staticmethod
-    def _check_bc_id_exists(rf_item):
-        return int_or_none(try_get(rf_item, lambda x: x['videos'][0]['url'])) is not None
-
-    def _entries(self, query, url):
-        query['size'] = 50
-        query['from'] = 0
-        for page_num in itertools.count(1):
-            results = self._call_api(
-                'search', None, query, url,
-                f'Downloading search JSON page {page_num}')
-            sl = try_get(results, lambda x: x['sectionList'][0], dict)
-            if sl:
-                results = sl
-            items = results.get('items')
-            if not items or not isinstance(items, list):
+    def _entries(self, payload):
+        from_val = 0
+
+        for page in itertools.count(1):
+            search = self._call_api(
+                'search', {**payload, 'from': from_val},
+                doseq=True, note=f'Downloading page {page}')
+            if not traverse_obj(search, 'sectionList'):
+                return
+
+            yield from [self.url_result(
+                f'{self._BASE_URL}/on-demand/on-demand-library.html#/session/{session_id}', CiscoLiveSessionIE)
+                for session_id in traverse_obj(search, (
+                    'sectionList', ..., 'items', ..., 'sessionID', {str_or_none}, filter, all, filter))]
+
+            from_val += int_or_none(payload['size'])
```

Comment:
`int_or_none` isn't doing any good here

either we let it be fatal
```suggestion
            from_val += int(payload['size'])
```
&nbsp;
or we take care not to be
```suggestion
            size = traverse_obj(payload, ('size', {int_or_none}))
            if not size:
                return
            from_val += size
```